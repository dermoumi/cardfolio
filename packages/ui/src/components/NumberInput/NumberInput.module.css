.number-input {
  display: block;
  padding: 0.5rem 1rem;

  font-family: var(--font-family-sans);
  font-size: var(--font-size-md);

  background-color: var(--color-surface);
  color: var(--color-text);

  border-radius: 999999px;
  border: 1px solid var(--color-primary);
}
