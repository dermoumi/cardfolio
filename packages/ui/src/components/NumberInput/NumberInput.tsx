import type { FC } from "react";

import styles from "./NumberInput.module.css";

export type NumberInputProps = {
  name: string;
  value?: number;
  onChange?: (value: number) => void;
  placeholder?: string;
  required?: boolean;
  min?: number;
  max?: number;
  form?: string;
};

const NumberInput: FC<NumberInputProps> = (
  { value, name, onChange, placeholder, required, min, max, form },
) => {
  return (
    <input
      className={styles.numberInput}
      name={name}
      type="number"
      value={value}
      onChange={(e) => onChange?.(Number(e.target.value))}
      placeholder={placeholder}
      required={required}
      min={min}
      max={max}
      form={form}
    />
  );
};

export default NumberInput;
