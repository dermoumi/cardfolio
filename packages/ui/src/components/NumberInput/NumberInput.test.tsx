import { fireEvent, render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import NumberInput from ".";

describe("NumberInput", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<NumberInput name="test" placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onChange when text is entered", () => {
    const handleChange = vi.fn();
    const { getByPlaceholderText } = render(
      <NumberInput name="test" placeholder="Enter text" onChange={handleChange} />,
    );

    const input = getByPlaceholderText("Enter text") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "144" } });

    expect(handleChange).toHaveBeenCalledWith(144);
  });
});
