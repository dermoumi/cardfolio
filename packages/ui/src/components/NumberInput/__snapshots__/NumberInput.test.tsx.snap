// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NumberInput > matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="_numberInput_1ad6bf"
    name="test"
    placeholder="Enter text"
    type="number"
  />
</DocumentFragment>
`;
